src/function/execute.rs

```diff
@@ -223,12 +229,14 @@ where
                     tracing::debug!(
                         "{database_key_index:?}: execute: iterate again, revisions: {revisions:#?}"
                     );
-                    opt_last_provisional = Some(self.insert_memo(
-                        zalsa,
-                        id,
-                        Memo::new(Some(new_value), zalsa.current_revision(), revisions),
-                        memo_ingredient_index,
-                    ));
+                    let memo = Memo::new(
+                        Some(new_value),
+                        zalsa.current_revision(),
+                        // TODO: Why is this correct?
```

Comment:
I think this is always correct, but I don't know enough about how cycles work to know why aren't calling `diff_outputs` before creating this memo. Could there be old values in `tracked_struct_ids` here? If so, we are now counting them as output dependencies, whereas previously I think they would just waste memory.